feat: align button

sharmadhiraj86 opened the pull request:
<!-- Insert GitHub issue number below -->
Fixes #1940 

<!-- Select which type of PR this is -->
- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
<!-- Briefly describe the proposed changes -->
The proposed changes align the button to the center in the case of loading button .
## Notes
<!-- Be sure to mention anything unusual, out-of-scope or new technical debt, etc -->

## Screenshots
<!-- Upload screenshots of UI before & after these changes -->

## Checklist
<!-- Acknowledge completion of steps in checklists below. Delete lists that are not applicable -->

<!-- For all PR types -->
- [x] I verify the build is in a non-broken state
- [ ] I verify all changes are within scope of the linked issue

<!-- For CSS changes -->
- [ ] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [ ] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

agliga commented:
Thanks for you contribution
Couple of issues:

First you shouldn't be modifing the `dist` files. Those are auto generated. You should modify the `src` less files and then run `npm run build` to generate the dist files. 
Second, that will be modifying any `form` buttons as well. This will break other non loading components. 
Third, can you add a storybook example for this if there isn't one?

sharmadhiraj86 commented:
- I have made the corresponding changes in the src file and removed the code that I have added previously.

- Can you let me know where all I need to validate if there is any changes.

- Can you let me know how I can add a story book example.

agliga commented:
> * I have made the corresponding changes in the src file and removed the code that I have added previously in the dist files.
So you need to add the dist files too. But you need to run `npm run build` so that it will generate the dist files. Please run that so that it compiles correctly.
> * Can you let me know where all I need to validate if there is any changes.
Yes, please launch the page using `npm start` and `npm run storybook` in order to see the changes
> * Can you let me know how I can add a story book example. @agliga
Sure, you should edit the `button.stories.js` file and add an example of what needs to be changed.


FYI, this change is breaking our existing code. If you want to fix loading button, you need to target that specifically. Start with the usecase in storybook. Build it out and then fix for that only.

If I have a 400px form button, this is what it should look like:
<img width="553" alt="Screen Shot 2022-12-09 at 10 16 54 AM" src="https://user-images.githubusercontent.com/1755269/206766917-23b706e2-def5-4143-b3ed-2e8f7a2586cb.png">

With your change, it looks like this:
<img width="583" alt="Screen Shot 2022-12-09 at 10 18 15 AM" src="https://user-images.githubusercontent.com/1755269/206767017-18a05ab5-7a88-4dfc-b4d5-cad7eaa5fdf5.png">

So you need to make sure it is fixing the issue but not breaking other components.

agliga commented:
Closing, we have a fix for this already. Thanks for the contribution anyways.
https://github.com/eBay/skin/pull/1955
